meson.build

```diff
@@ -0,0 +1,169 @@
+project('xviewer', 'c', version: '3.2.1', meson_version: '>= 0.49')
+
+version = meson.project_version()
+
+xviewer_conf = configuration_data()
+xviewer_conf.set_quoted('VERSION', version)
+xviewer_conf.set_quoted('GETTEXT_PACKAGE', meson.project_name())
+xviewer_conf.set('HAVE_INTROSPECTION', 1)
+
+# directories
+prefix = get_option('prefix')
+datadir = prefix / get_option('datadir')
+libdir = prefix / get_option('libdir')
+localedir = prefix /  get_option('localedir')
+
+xviewer_pkgdatadir = datadir / meson.project_name()
+xviewer_pkglibdir = libdir / meson.project_name()
+# workaround for plugins: keep old lib dir without arch-triplet
+xviewer_pluginsdir = prefix / 'lib' /  meson.project_name() / 'plugins'
+
+xviewer_conf.set_quoted('LIBDIR', libdir)
+xviewer_conf.set_quoted('XVIEWER_DATA_DIR', xviewer_pkgdatadir)
+xviewer_conf.set_quoted('XVIEWER_LOCALE_DIR', localedir)
+xviewer_conf.set_quoted('XVIEWER_PLUGIN_DIR', xviewer_pluginsdir)
+
+src_root = meson.source_root()
+po_dir = join_paths(src_root, 'po')
+
+# options
+enable_docs = get_option('docs')
+enable_exempi = get_option('enable_exempi')
+enable_exif = get_option('enable_exif')
+enable_jpeg = get_option('enable_jpeg')
+enable_lcms = get_option('enable_lcms')
+enable_rsvg = get_option('enable_rsvg')
+
+gnome = import('gnome')
+i18n = import('i18n')
+pkgconfig = import('pkgconfig')
+
+# dependencies
+glib = dependency('glib-2.0', version: '>= 2.38.0')
+gtk = dependency('gtk+-3.0')
+libpeas = dependency('libpeas-1.0', version: '>= 0.7.4')
+libpeas_gtk = dependency('libpeas-gtk-1.0', version: '>= 0.7.4')
+pixbuf = dependency('gdk-pixbuf-2.0', version: '>= 2.19.1')
+X11 = dependency('x11')
+zlib = dependency('zlib')
+cinnamon_desktop = dependency('cinnamon-desktop', version: '>= 3.2.0')
+xapp = dependency('xapp', version: '>= 1.9.0')
+
+intltool_merge = find_program('intltool-merge')
+cc = meson.get_compiler('c')
+
+# on some systems we need to find the math lib to make sure it builds
+math = cc.find_library('m')
+
+# taken from eog's meson.build
+# support for nl_langinfo (_NL_MEASUREMENT_MEASUREMENT) (optional)
+langinfo_measurement_src = '''
+  #include <langinfo.h>
+  int main() {
+    char c;
+    c = *((unsigned char *)  nl_langinfo(_NL_MEASUREMENT_MEASUREMENT));
+  };
+'''
+xviewer_conf.set('HAVE__NL_MEASUREMENT_MEASUREMENT', cc.compiles(langinfo_measurement_src),
+             description: 'Define if _NL_MEASUREMENT_MEASUREMENT is available')
+
+# support for strptime
+xviewer_conf.set('HAVE_STRPTIME', cc.has_function('strptime'))
+
+# (optional) XMP support
+if enable_exempi
+    exempi = dependency('exempi-2.0', version: '>= 1.99.5')
+    xviewer_conf.set10('HAVE_EXEMPI', true)
```

Comment:
... and use them here.

```meson
exempi = dependency('exempi-2.0', version: '>= 1.99.5', required: get_option('exempi'))
if exempi.found()
    xviewer_conf.set10('HAVE_EXEMPI', true)
endif
```

This lets you specify enabled/disabled or *auto*, and, in the disabled case, it actually logs a helpful status message:

```
Dependency exempi skipped: feature exempi disabled
```